doc/tutorial/intro-to-autoinstall.rst

```diff
@@ -0,0 +1,243 @@
+
+.. code-block:: yaml
+
+    #cloud-config
+    autoinstall:
+        version: 1
+        ....
+
+Autoinstall on the install media
+--------------------------------
+
+Another option for supplying autoinstall to the Ubuntu installer is to place a
+file named :code:`autoinstall.yaml` on the install media itself.
+
+There are two potential locations for the :code:`autoinstall.yaml` file:
+ * At the root of the "cdrom".  When writing the installation ISO to a USB
+   Flash Drive, this can be done by copying the :code:`autoinstall.yaml` to the
+   partition containing the contents of the ISO - i.e., in the same directory
+   containing the :code:`casper` directory.
+ * On the rootfs of the installation system - this option will typically
+   require modifying the installation ISO and is not suggested, but is
+   supported.
+
+Directly specifying autoinstall as a :code:`autoinstall.yaml` file does not
+require a :code:`#cloud-config` header, and does not use a top level
+:code:`autoinstall` key.  The autoinstall directives are placed at the top
+level.
+
+.. code-block:: yaml
+
+    version: 1
+    ....
+
+
+Cloud-init and Autoinstall interaction
+======================================
+
+While cloud-init may assist in providing the autoinstall configuration to the
+Ubuntu installer, cloud-init itself is not processing the autoinstall.
+
+If cloud-init directives are intended to modify the ephemeral system, they
+must appear at the top level of the cloud-config.  If instead
+cloud-init directives are intended to modify the system being installed, they
+must appear under a :code:`user-data` section in :code:`autoinstall`.
+
+.. code-block:: yaml
+
+    #cloud-config
+    # cloud-init directives may optionally be specified here.
+    # These directives affect the ephemeral system performing the install.
+
+    autoinstall:
+        # autoinstall directives must be specified here, not directly at the
+        # top level.  These directives are processed by the Ubuntu Installer,
+        # and configure the target system to be installed.
+
+        user-data:
+            # cloud-init directives may also be optionally be specified here.
+            # These directives also affect the target system to be installed,
+            # and are processed on first boot.
+
+
+Zero-touch deployment with autoinstall
+======================================
+
+The Ubuntu Installer contains a safeguard, intended to prevent USB Flash Drives
+with an :code:`autoinstall.yaml` file from wiping out the wrong system.
+
+During autoinstall, a prompt will be shown to confirm that the install really
+should proceed and start making modifications to the target system. ::
+
+    start: subiquity/Meta/status_GET
+    Confirmation is required to continue.
+    Add 'autoinstall' to your kernel command line to avoid this
+
+
+    Continue with autoinstall? (yes|no)
+
+To bypass this prompt, arrange for the argument :code:`autoinstall` to be
```

Comment:
Would the typical user know how to do this? If not, may need to include a short how-to guide

Reply:
Right, I agree and think this is poorly placed in the tutorial doc as is but wanted to give it a home before we moved stuff around.

We need a proper tutorial, which I intend to make qemu based.  As part of that we can show the process for kernel command line overrides.  This note in particular more feels like a reference bit but lacks a good spot as of yet.  I'm thinking some sort of secondary reference document alongside but different to the "Autoinstall configuration reference manual", as this isn't a directive.

Reply:
That sounds great! Since it's on the to-do list, it's fine to mark this as resolved.